modules/host/network.go

```diff
@@ -49,11 +49,11 @@ func (h *Host) initNetworking(address string) (err error) {
 		return err
 	}
 	// Automatically close the listener when h.tg.Stop() is called.
+	err = h.tg.Add()
+	if err != nil {
+		return err
+	}
 	go func() {
-		err := h.tg.Add()
-		if err != nil {
-			return
-		}
```

Comment:
Host tests were timing out non-deterministically on Close. I believe this was the cause. It might not be a deadlock in the original code (ungrouped `listener.Close` and `listener.Accept`) if the execution order of goroutines is guaranteed to be the same as the calling order, but I'm not sure this is true.